Feature/careplan start date

achen2401 opened the pull request:
https://www.pivotaltracker.com/n/projects/2584667

- allow start date to be specified for a CarePlan
- date/time of scheduled messages will be calculated based on the specified start date (default was today's date/time)

(Noting that the code for`EnrollmentApp.tsx` has been reformatted by my vscode editor so appeared to have more changes than there were.  The primary change is the new method, `getCarePlanStartDateModalView` added to display the modal)

screenshot of modal:
![Screenshot 2024-06-07 at 12 27 15 PM](https://github.com/uwcirg/isacc-messaging-client-sof/assets/12942714/1bd8e0c8-5fc3-41d2-935e-1e0d40390bd6)

Commit:
allow specification of careplan start date

Commit:
remove debug statement, add new line

mcjustin commented:
Thanks for the quick work on this @achen2401 . 
1. I was looking for the string "Please Specify the CarePlan Start Date" in this MR, but I'm not finding it... is there another file that you need to include here? 
2. I was looking for that because it should be sentence-case, like the rest of the website, i.e. "Please specify the CarePlan start date". Same for "Start Date & Time" > "Start date & time".

Commit:
fix text casings per feedback

achen2401 commented:
> 
> 1. I was looking for the string "Please Specify the CarePlan Start Date" in this MR, but I'm not finding it... is there another file that you need to include here?
> 2. I was looking for that because it should be sentence-case, like the rest of the website, i.e. "Please specify the CarePlan start date". Same for "Start Date & Time" > "Start date & time".

thank you @mcjustin I fixed the text and updated the screenshot.  The text was referenced [here](https://github.com/uwcirg/isacc-messaging-client-sof/blob/feature/careplan-start-date/src/components/EnrollmentApp.tsx#L264) in `EnrollmentApp.tsx`

mcjustin commented:
> > 1. I was looking for the string "Please Specify the CarePlan Start Date" in this MR, but I'm not finding it... is there another file that you need to include here?
> > 2. I was looking for that because it should be sentence-case, like the rest of the website, i.e. "Please specify the CarePlan start date". Same for "Start Date & Time" > "Start date & time".
> 
> thank you @mcjustin I fixed the text and updated the screenshot. The text was referenced [here](https://github.com/uwcirg/isacc-messaging-client-sof/blob/feature/careplan-start-date/src/components/EnrollmentApp.tsx#L264) in `EnrollmentApp.tsx`

Thanks - sorry, I probably didn't expand some that file in the diff view.
